service/src/main/java/bio/terra/workspace/common/logging/WorkspaceActivityLogHook.java

```diff
@@ -247,37 +247,37 @@ private void maybeLogFolderDeletion(
     }
   }
 
-  private void maybeLogControlledResourceDeletion(
+  private void maybeLogControlledResourcesDeletionFlight(
       FlightContext context, UUID workspaceUuid, String userEmail, String subjectId) {
-    UUID resourceId = getControlledResourceToDeleteFromFlight(context);
-    try {
-      resourceDao.getResource(workspaceUuid, resourceId);
-      logger.warn(
-          String.format(
-              "Controlled resource %s in workspace %s is failed to be deleted; "
-                  + "not writing deletion to workspace activity log",
-              resourceId, workspaceUuid));
-    } catch (ResourceNotFoundException e) {
-      activityLogDao.writeActivity(
-          workspaceUuid,
-          new DbWorkspaceActivityLog(
-              userEmail,
-              subjectId,
-              OperationType.DELETE,
-              resourceId.toString(),
-              ActivityLogChangedTarget.RESOURCE));
+    List<UUID> resourceIds = getControlledResourceToDeleteFromFlight(context);
+    for (var resourceId : resourceIds) {
+      try {
+        resourceDao.getResource(workspaceUuid, resourceId);
+        logger.warn(
+            String.format(
+                "Controlled resource %s in workspace %s is failed to be deleted; "
+                    + "not writing deletion to workspace activity log",
+                resourceId, workspaceUuid));
+      } catch (ResourceNotFoundException e) {
+        activityLogDao.writeActivity(
+            workspaceUuid,
+            new DbWorkspaceActivityLog(
+                userEmail,
+                subjectId,
+                OperationType.DELETE,
+                resourceId.toString(),
+                ActivityLogChangedTarget.RESOURCE));
+      }
     }
   }
 
-  private UUID getControlledResourceToDeleteFromFlight(FlightContext context) {
+  private List<UUID> getControlledResourceToDeleteFromFlight(FlightContext context) {
```

Comment:
all the controlled resource deletion takes in a list now https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/service/resource/controlled/ControlledResourceService.java#L627.

I made the change because i wanted to reuse the same flight for multi-resource deletion (e.g. a flight to delete a folder).